### DISCUSSION TITLE
Should we support `kotlinx-coroutines-test`?

### DISCUSSION THREAD

**Discussion by BraisGabin** [detekt/detekt#4642] [General]
Right now we have rules related with `kotlinx-coroutines-core` but I would like to add a rule that is 100% related with `kotlinx-coroutines-test` (the usages of `runBlockingTest` and `runTest`). A rule like that would fit detekt or should I create it as a custom rule?

**Comment by marschwar** [comment]
I don't see why this should be a problem. Go for it.

**Comment by cortinico** [comment]
I think this is a bit borderline. We do have rules for `coroutines` as that is a language feature. `kotlinx-coroutines-core` is technically an external library, but makes coroutines unusable without it.

`kotlinx-coroutines-test` goes one step further and makes me think that it should stay in a separate custom rule. Similarly we don't have tests for `kotlin-test`.

But that's up for debate I guess.

**Reply by marschwar** [reply]
You are probably right. It is really hard to know where to draw the line. For the official coroutines testing library I would make an exception, but then it becomes hard to argue against rules for `kotlinx-coroutines-reactor` or `kotlinx-coroutines-play-services` and they definitely do not feel right.

**Reply by BraisGabin** [reply]
I agree too. `kotlinx-coroutines-play-services` wtf?

I will create those rules as custom rules ;)
